pkg/gofr/migration/sql_test.go

```diff
@@ -4,123 +4,153 @@ import (
 	t.Run("successful query row", func(t *testing.T) {
-		ctrl := gomock.NewController(t)
-		mockDB := container.NewMockDB(ctrl)
-		expectedRow := &sql.Row{}
+		var name string
 
-		mockDB.EXPECT().QueryRow("SELECT * FROM users WHERE id = ?", 1).Return(expectedRow)
-		sqlMockDB := mockDB
+		var id int
 
-		row := sqlMockDB.QueryRow("SELECT * FROM users WHERE id = ?", 1)
+		mockContainer, mocks := container.NewMockContainer(t)
 
-		if row != expectedRow {
-			t.Errorf("QueryRow should return the expected row, got: %v", row)
-		}
+		expectedRows := mocks.SQL.NewRows([]string{"id", "name"}).AddRow(1, "Alex")
+
+		mocks.SQL.ExpectQuery("SELECT * FROM users WHERE id = ?").WithArgs(1).WillReturnRows(expectedRows)
+		sqlMockDB := mockContainer.SQL
+
+		err := sqlMockDB.QueryRow("SELECT * FROM users WHERE id = ?", 1).Scan(&id, &name)
+		require.NoError(t, err, "Error while scanning row")
+		assert.Equal(t, 1, id, "expected id to be equal to 1")
+		assert.Equal(t, "Alex", name, "expected name to be equal to 'Alex'")
 	})
 }
 
 func TestQueryRowContext(t *testing.T) {
 	ctx := context.Background()
 
 	t.Run("successful query row context", func(t *testing.T) {
-		ctrl := gomock.NewController(t)
-		mockDB := container.NewMockDB(ctrl)
-		expectedRow := &sql.Row{}
-		mockDB.EXPECT().QueryRowContext(ctx, "SELECT * FROM users WHERE id = ?", 1).Return(expectedRow)
-		sqlMockDB := mockDB
+		var id int
 
-		row := sqlMockDB.QueryRowContext(ctx, "SELECT * FROM users WHERE id = ?", 1)
+		var name string
 
-		if row != expectedRow {
-			t.Errorf("QueryRowContext should return the expected row,  got: %v", row)
-		}
+		mockContainer, mocks := container.NewMockContainer(t)
+
+		expectedRows := mocks.SQL.NewRows([]string{"id", "name"}).AddRow(1, "Alex")
+
+		mocks.SQL.ExpectQuery("SELECT * FROM users WHERE id = ?").WithArgs(1).WillReturnRows(expectedRows)
+		sqlMockDB := mockContainer.SQL
+
+		err := sqlMockDB.QueryRowContext(ctx, "SELECT * FROM users WHERE id = ?", 1).Scan(&id, &name)
+		require.NoError(t, err, "Error while scanning row")
+		assert.Equal(t, 1, id, "expected id to be equal to 1")
+		assert.Equal(t, "Alex", name, "expected name to be equal to 'Alex'")
 	})
 }
 
 func TestExec(t *testing.T) {
 	t.Run("successful exec", func(t *testing.T) {
-		ctrl := gomock.NewController(t)
-		mockDB := container.NewMockDB(ctrl)
-		expectedResult := sqlmock.NewResult(10, 1)
+		mockContainer, mocks := container.NewMockContainer(t)
+
+		expectedResult := mocks.SQL.NewResult(10, 1)
 
-		mockDB.EXPECT().Exec("DELETE FROM users WHERE id = ?", 1).Return(expectedResult, nil)
-		sqlDB := mockDB
+		mocks.SQL.ExpectExec("DELETE FROM users WHERE id = ?").WithArgs(1).WillReturnResult(expectedResult)
+		sqlDB := mockContainer.SQL
 
 		result, err := sqlDB.Exec("DELETE FROM users WHERE id = ?", 1)
+		require.NoError(t, err)
 
-		if err != nil {
-			t.Errorf("Exec should return no error, got: %v", err)
-		}
+		expectedLastInserted, err := expectedResult.LastInsertId()
```

Comment:
a last insertid on a delete ?

I understand everything is mocked but it's strange

Reply:
I suppose Last insertID does not change with a delete query...but as all exec commands return Result that has last inserted and rowsAffected .....I think a non-zero value would be returned. I might be wrong....but I can't find any definite answer about this, around on the internet.